File: app/src/main/java/com/s95ammar/weeklyschedule/views/adapters/CategorySpinnerAdapter.java
package com.s95ammar.weeklyschedule.views.adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.graphics.drawable.DrawableCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.s95ammar.weeklyschedule.R;
import com.s95ammar.weeklyschedule.models.Category;

import java.util.ArrayList;

public class CategorySpinnerAdapter extends ArrayAdapter<Category> {

    public CategorySpinnerAdapter(Context context, ArrayList<Category> categoriesList) {
        super(context, 0, categoriesList);
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        return initView(position, convertView, parent);
    }

    @Override
    public View getDropDownView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        return initView(position, convertView, parent);
    }

    private View initView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(
                    R.layout.category_spinner_row, parent, false
            );
        }

        TextView textViewName = convertView.findViewById(R.id.textView_category_spinner);

        Category currentItem = getItem(position);

        if (currentItem != null) {
            textViewName.setText(currentItem.getName());
            textViewName.setTextColor(currentItem.getTextColor());
            DrawableCompat.setTint(textViewName.getBackground(), currentItem.getFillColor());

        }

        return convertView;
    }
}

File: app/src/main/java/com/s95ammar/weeklyschedule/views/adapters/CategorySpinnerAdapter.kt
package com.s95ammar.weeklyschedule.views.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import androidx.annotation.Nullable
import com.s95ammar.weeklyschedule.R
import com.s95ammar.weeklyschedule.models.data.Category
import com.s95ammar.weeklyschedule.util.getStrokedBackground
import kotlinx.android.synthetic.main.spinner_row_category.view.*


class CategorySpinnerAdapter(context: Context, categoriesList: List<Category>) : ArrayAdapter<Category>(context, 0, categoriesList) {

	override fun getView(position: Int, @Nullable convertView: View?, parent: ViewGroup): View {
		return initView(position, convertView, parent)
	}

	override fun getDropDownView(position: Int, @Nullable convertView: View?, parent: ViewGroup): View {
		return initView(position, convertView, parent)
	}

	private fun initView(position: Int, convertView: View?, parent: ViewGroup): View {
		val view = convertView ?: LayoutInflater.from(context).inflate(R.layout.spinner_row_category, parent, false)
		getItem(position)?.let { category ->
			view.textView_category_spinner.apply {
				text = category.name
				setTextColor(category.textColor)
				background = getStrokedBackground(category.fillColor, category.textColor)
			}
		}
		return view
	}
}
